src/librustc_mir/transform/cleanup_post_borrowck.rs

```diff
@@ -93,7 +93,33 @@ impl<'a, 'tcx> MutVisitor<'tcx> for DeleteTrivialEndRegions<'a> {
         }
 
         if delete_it {
-            statement.kind = StatementKind::Nop;
+            statement.make_nop();
+        }
+        self.super_statement(block, statement, location);
+    }
+}
+
+pub struct CleanUserAssertTy;
```

Comment:
why are these two distinct things?

Reply:
If you are referring to the separation of the passes, in my experimentation locally, I found that the passes had to happen at different times otherwise the `UserAssertTy` statement wouldn't be in the MIR when the NLL type check code gets it (since the type check pass doesn't run when `#![feature(nll)]` was used - as far as I could tell - ordering it relative to that wasn't viable). Therefore they had to be separate. If this doesn't sound right, I can take another look.

Reply:
Hmm. I didn't realize that these passes were running at distinct times. At minimum, could use a comment clarifying what is happening here =)

Reply:
I'll add one with my next push.

Reply:
This discussion should be resolved - I updated the comment on the module to mention this. It doesn't affect these lines specifically so this isn't marked as "outdated".